src/com/facebook/buck/jvm/kotlin/KotlinBuckConfig.java

```diff
@@ -166,23 +143,30 @@ Path getPathToScriptRuntimeJar() {
    * @return the Kotlin compiler jar path
    */
   Path getPathToCompilerJar() {
-    Path compiler = getKotlinHome().resolve("kotlin-compiler.jar");
-    if (Files.isRegularFile(compiler)) {
-      return compiler.normalize();
-    }
-
-    compiler = getKotlinHome().resolve(Paths.get("lib", "kotlin-compiler.jar"));
-    if (Files.isRegularFile(compiler)) {
-      return compiler.normalize();
+    try {
+      return getPathToJar("kotlin-compiler-embeddable");
+    } catch (HumanReadableException e) {
+      LOG.warn("kotlin-compiler-embeddable.jar was not found in " + kotlinHome + " directory, this"
+          + " may result in kapt not working properly. Proceeding with kotlin-compiler.jar"
+      );
+      return getPathToJar("kotlin-compiler");
     }
+  }
 
-    compiler = getKotlinHome().resolve(Paths.get("libexec", "lib", "kotlin-compiler.jar"));
-    if (Files.isRegularFile(compiler)) {
-      return compiler.normalize();
+  /**
+   * Get the path to the Kotlin annotation processing jar.
+   *
+   * @return the Kotlin annotation processing jar path
+   */
+  Path getPathToAP() {
```

Comment:
No abbreviations, please. By the name I would have expected this to be getting the path to the annotation processor itself, and that's not what it's doing.